### PR TITLE
Internal: Remove Strings.cleanPath

### PR DESCRIPTION
**Pull request by rjernst**
This commit removes the cleanPath method, in favor of using java's
Path.normalize().

### CODE REVIEW

**Review by talevy — APPROVED** [review]
LGTM

File: core/src/main/java/org/elasticsearch/common/Strings.java [hunk 1]
```diff
@@ -404,66 +404,6 @@ public static boolean validFileNameExcludingAstrix(String fileName) {
...
-                }
-            }
-        }
-
-        // Remaining top paths need to be retained.
-        for (int i = 0; i < tops; i++) {
-            pathElements.add(0, TOP_PATH);
-        }
-
-        return prefix + collectionToDelimitedString(pathElements, FOLDER_SEPARATOR);
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/common/Strings.java]: I think that `FOLDER_SEPARATOR` and `WINDOWS_FOLDER_SEPARATOR` are dead fields now?

**Review by jasontedor — APPROVED** [review]
I left one comment. Another is that there is a behavior difference between `Strings#cleanPath` and `Path#normalize`; namely the former was converting Windows folder separators to Unix folder separators. I do not think that should matter anywhere (if it did, we can fix it), just something to keep in mind. I'm glad to see this method gone, it never made sense, especially living in `org.elasticsearch.common.Strings`.

File: core/src/main/java/org/elasticsearch/common/Strings.java (hunk 1 above)
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/common/Strings.java]: Removed.

**Pull request merged**